flurry-provider/src/main/java/com/busybusy/flurry_provider/FlurryProvider.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2016 - 2022 busybusy, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.busybusy.flurry_provider
+
+import com.busybusy.analyticskit_android.AnalyticsKitProvider.PriorityFilter
+import com.busybusy.analyticskit_android.AnalyticsKitProvider
+import com.busybusy.analyticskit_android.AnalyticsEvent
+import com.flurry.android.FlurryAgent
+import java.lang.IllegalStateException
+
+/**
+ * Implements Flurry as a provider to use with [com.busybusy.analyticskit_android.AnalyticsKit]
+ *
+ *
+ * **Important**: It is a violation of Flurry’s TOS to record personally identifiable information such as a user’s UDID,
+ * email address, and so on using Flurry. If you have a user login that you wish to associate with your session and
+ * event data, you should use the SetUserID function. If you do choose to record a user id of any type within a parameter,
+ * you must anonymize the data using a hashing function such as MD5 or SHA256 prior to calling the method.
+ *
+ *
+ * Refer to the Flurry documentation here:
+ * [Flurry Documentation](https://developer.yahoo.com/flurry/docs/analytics/gettingstarted/events/android/)
+ *
+ * @author John Hunt on 3/21/16.
+ */
+class FlurryProvider(
+    private var priorityFilter: PriorityFilter = PriorityFilter { true },
+) : AnalyticsKitProvider {
+
+    val ATTRIBUTE_LIMIT = 10
+
+    /**
+     * Returns the filter used to restrict events by priority.
+     *
+     * @return the [PriorityFilter] instance the provider is using to determine if an
+     *         event of a given priority should be logged
+     */
+    override fun getPriorityFilter(): PriorityFilter = priorityFilter
+
```

Comment:
Removed the public `setPriorityFilter` function. Justification provided in [this comment](https://github.com/busybusy/AnalyticsKit-Android/pull/29#discussion_r799166791).